### DISCUSSION TITLE
Decreasing convergence for Darcy equation

### DISCUSSION THREAD

**Discussion by AdamG412** [wenceorg/comp4187#88] [Q&A]
I am currently having an issue with part 3 solving the stationary Darcy equation. Using the test_mms_convergence function from test_one_dim with the LU solver, the values of `np.log2(errors[:-1] / errors[1:]` are 
`[2.64595681 1.95218595 1.63508536 1.53558265]`
And using the multigrid solver they are
`[2.3432822  2.15815936 1.87067679 1.68959604]`
in both cases starting with a grid of size 2 and refining up to 5 times.
As far as I can tell the errors aren't occurring on the edge of the domain, and I am scaling both the left and right hand sides by Hx*Hy*Hz. Are there any areas I might be missing?

**Comment by annereinarz** [comment]
The volume scaling factor is only needed for the multigrid solver, if you have the same convergence issues with the LU decomposition then the problem must be in your matrix, rhs or error computation.
In the 1D case the l2 norm of the error is computed as: Hx*numpy.linalg.norm(error), I assume you have the appropriate 3D scaling factor in your error norm?

**Reply by AdamG412** [reply]
yes, I have Hx*Hy*Hz*np.linalg.norm(error)

**Reply by annereinarz** [reply]
A few ideas to try to pin down the source of the error

1. What are you using as the expected solution, are you sure this is correct?
2. Is the right hand side (rhs) correctly scaled? You can try to check this by using an identity operator instead of your current laplacian. The function on the rhs should then converge against f. This essentially corresponds to applying an interpolation operator.
3. Do the same thing for the operator. Check the scaling of the matrix with a rhs of 1 (make sure the Dirichlet conditions are set to 1 as well)
